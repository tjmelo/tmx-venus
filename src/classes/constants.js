export const KEYS = {
  API: 'https://63f239144f17278c9a22d1cd.mockapi.io/api/v1/item',
};

export const ELEM = {
  ADD_ITEM: document.querySelector('.add-item'),
  ALL_ITEMS: document.querySelector('.all-items'),
};
