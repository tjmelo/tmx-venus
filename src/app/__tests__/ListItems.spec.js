import toListItems from '../modules/ListItems';
import toFetchAPI from '../modules/fetchAPI';

import { KEYS } from '../constants';

jest.mock('../modules/fetchAPI');
jest.mock('../modules/ListItems');

describe('Should to list the sequence items in app', () => {
  it('Should request toFetchAPI for get list items by API', () => {
    const testToFetchAPI = jest.fn();
    testToFetchAPI.mockImplementation((argTest) => toFetchAPI(argTest));

    // do
    testToFetchAPI(KEYS.API);

    // then
    expect(testToFetchAPI).toHaveReturned();
    expect(testToFetchAPI).toHaveBeenCalledTimes(1);
    expect(testToFetchAPI).toHaveBeenCalledWith(KEYS.API);

    // Expects fidelity with scope of function
    expect(testToFetchAPI).toMatchSnapshot();
  });

  it('Should request list of items', () => {
    const testToListItems = jest.fn();
    testToListItems.mockImplementation(() => toListItems());

    // do
    testToListItems();

    // then
    expect(testToListItems).toHaveReturned();
    expect(testToListItems).toHaveBeenCalledTimes(1);

    // Expects fidelity with scope of function
    expect(testToListItems).toMatchSnapshot();
  });
});
