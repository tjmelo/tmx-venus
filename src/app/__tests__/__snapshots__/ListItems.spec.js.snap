// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should to list the sequence items in app Should request list of items 1`] = `
[MockFunction] {
  "calls": [
    [],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Should to list the sequence items in app Should request toFetchAPI for get list items by API 1`] = `
[MockFunction] {
  "calls": [
    [
      "https://63f239144f17278c9a22d1cd.mockapi.io/api/v1/item",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
